nextjs-frontend/app/dashboard/add-item/page.tsx

```diff
@@ -0,0 +1,105 @@
+"use client";
+
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { addItem } from "@/components/actions/items-action";
+import { useActionState } from "react";
+import { SubmitButton } from "@/components/ui/submitButton";
+
+const initialState = { message: "" };
+
+export default function CreateItemPage() {
+  const [state, dispatch] = useActionState(addItem, initialState);
+
+  return (
+    <div className="bg-gray-50 dark:bg-gray-900 min-h-screen">
+      <div className="max-w-4xl mx-auto p-6">
+        <header className="mb-6">
+          <h1 className="text-3xl font-semibold text-gray-800 dark:text-white">
+            Create New Item
+          </h1>
+          <p className="text-lg text-gray-600 dark:text-gray-400">
+            Enter the details of the new item below.
+          </p>
+        </header>
+
+        <form
+          action={dispatch}
+          className="bg-white dark:bg-gray-800 rounded-lg shadow-lg p-8 space-y-6"
+        >
+          <div className="space-y-6">
+            <div className="space-y-3">
+              <Label
+                htmlFor="name"
+                className="text-gray-700 dark:text-gray-300"
+              >
+                Item Name
+              </Label>
+              <Input
+                id="name"
+                name="name"
+                type="text"
+                placeholder="Item name"
+                required
+                className="w-full border-gray-300 dark:border-gray-600"
+              />
+              {state.errors?.name && (
```

Comment:
Ideally those errors should be handled by the form library. You can use a form library that's compatible with zod. react-hook-form seems to be, so there should be a better error handling with a FormInput component you can make.

Reply:
I was following the nexts-js guidelines and they are asking to make the validations in the next.js server. But, when I worked with React we used to do the validations on the client components as you are suggesting.

I'll study this a little bit more, but at first, it seems redundant to me make the validations in both places and in the server seems more secure. Maybe on the client should be only validations that require a fast feedback. Or maybe should be in both places, but definitvely shoudl be validations in the server actions.

Reply:
In our case, the final server, the API, does all necessary validations. What I mean here is integrating better the validations and the components, I can clarify that.

Reply:
I mean server side validation (Next.js server): This is what we are doing, but I got your point:
https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations#server-side-form-validation